tests/adapters/git/test_github_git_provider.py

```diff
@@ -40,54 +44,88 @@ def test_get_user_info_failing(self, github_mock):
+        github_provider = GitHubGitProvider()
+
+        result = github_provider._get_repository("organization", "fake-project")
+
+        self.assertEqual(result, fake_repo)
+
+    def test_get_repository_on_non_existing_repository(self, github_mock):
+        """Should raise a GitResourceNotFoundError error."""
+        instance = github_mock.return_value
+        instance.get_repo.side_effect = GithubException(404, "fake error")
+        github_provider = GitHubGitProvider()
+
+        with self.assertRaises(GitResourceNotFoundError) as context:
+            github_provider._get_repository("organization", "fake-project")
+
+        self.assertEqual(context.exception.resource, GitResource.REPOSITORY)
+
+    def test_get_repository_failing(self, github_mock):
+        """Should raise a GitProviderError error."""
+        instance = github_mock.return_value
+        instance.get_repo.side_effect = GithubException(500, "fake error")
+        github_provider = GitHubGitProvider()
+
+        with self.assertRaises(GitProviderError):
+            github_provider._get_repository("organization", "fake-project")
+
+    @patch.object(GitHubGitProvider, "_get_repository", autospec=True)
+    def test_get_branch(self, get_repository_mock, _github_mock):
         """Should get the branch data."""
         # Mocks
         fake_branch_result = MagicMock(spec=Branch.Branch)
-        instance = github_mock.return_value
-        instance.get_repo.return_value.get_branch.return_value = fake_branch_result
+        get_repository_mock.return_value.get_branch.return_value = fake_branch_result
         github_provider = GitHubGitProvider()
 
         # Test
         result = github_provider.get_branch("organization", "fake-project", "fake-branch")
 
         # Assertions
-        instance.get_repo.assert_called_once_with("organization/fake-project")
-        instance.get_repo.return_value.get_branch.assert_called_once_with(branch="fake-branch")
+        get_repository_mock.assert_called_once_with(github_provider, "organization", "fake-project")
+        get_repository_mock.return_value.get_branch.assert_called_once_with(branch="fake-branch")
         self.assertEqual(result, fake_branch_result)
 
-    def test_get_branch_not_found(self, github_mock):
+    @patch.object(GitHubGitProvider, "_get_repository", autospec=True)
+    def test_get_branch_not_found(self, get_repository_mock, _github_mock):
         """Should return None when the branch is not found."""
         # Mocks
-        instance = github_mock.return_value
-        instance.get_repo.return_value.get_branch.side_effect = GithubException(
+        get_repository_mock.return_value.get_branch.side_effect = GithubException(
             404, "branch not found"
         )
         github_provider = GitHubGitProvider()
 
-        # Test
-        result = github_provider.get_branch("organization", "fake-project", "fake-branch")
+        # Tests
+        result = None
```

Comment:
I think asserting `result` is misleading since we expect the function to raise an error
```suggestion
```

Reply:
Yes forgot to remove this line ;)